app/sample/gpu_direct/tx_st20_pipeline_gpu_direct.c

```diff
@@ -0,0 +1,278 @@
+           MTL_IP_ADDR_LEN);
+    snprintf(ops_tx.port.port[MTL_SESSION_PORT_P], MTL_PORT_MAX_LEN, "%s",
+             ctx.param.port[MTL_PORT_P]);
+    ops_tx.port.udp_port[MTL_SESSION_PORT_P] = ctx.udp_port + i * 2;
+    if (ops_tx.port.num_port > 1) {
+      memcpy(ops_tx.port.dip_addr[MTL_SESSION_PORT_R], ctx.tx_dip_addr[MTL_PORT_R],
+             MTL_IP_ADDR_LEN);
+      snprintf(ops_tx.port.port[MTL_SESSION_PORT_R], MTL_PORT_MAX_LEN, "%s",
+               ctx.param.port[MTL_PORT_R]);
+      ops_tx.port.udp_port[MTL_SESSION_PORT_R] = ctx.udp_port + i * 2;
+    }
+    if (ctx.multi_inc_addr) {
+      /* use a new ip addr instead of a new udp port for multi sessions */
+      ops_tx.port.udp_port[MTL_SESSION_PORT_P] = ctx.udp_port;
+      ops_tx.port.dip_addr[MTL_SESSION_PORT_P][3] += i;
+    }
+    ops_tx.port.payload_type = ctx.payload_type;
+    ops_tx.width = ctx.width;
+    ops_tx.height = ctx.height;
+    ops_tx.fps = ctx.fps;
+    ops_tx.interlaced = ctx.interlaced;
+    ops_tx.input_fmt = ctx.input_fmt;
+    ops_tx.transport_fmt = ctx.fmt;
+    ops_tx.device = ST_PLUGIN_DEVICE_AUTO;
+    ops_tx.framebuff_cnt = ctx.framebuff_cnt;
+    ops_tx.flags = ST20P_TX_FLAG_BLOCK_GET;
+    ops_tx.notify_frame_done = tx_st20p_frame_done;
+    ops_tx.transport_packing = ctx.packing;
+
+    st20p_tx_handle tx_handle = st20p_tx_create(ctx.st, &ops_tx);
+    if (!tx_handle) {
+      err("%s(%d), st20p_tx_create failed\n", __func__, i);
+      ret = -EIO;
+      goto error;
+    }
+    app[i]->handle = tx_handle;
+
+    app[i]->frame_size = st20p_tx_frame_size(tx_handle);
+    info("%s(%d), frame_size %" PRId64 "\n", __func__, i, app[i]->frame_size);
+    ret = tx_st20p_open_source(app[i], ctx.tx_url);
+    if (ret < 0) {
+      err("%s(%d), open source failed\n", __func__, i);
+      goto error;
+    }
+
+    ret = pthread_create(&app[i]->frame_thread, NULL, tx_st20p_frame_thread, app[i]);
+    if (ret < 0) {
+      err("%s(%d), thread create failed %d\n", __func__, ret, i);
+      ret = -EIO;
+      goto error;
+    }
+  }
+
+  while (!ctx.exit) {
+    sleep(1);
+  }
+
+  // stop app thread
+  for (int i = 0; i < session_num; i++) {
+    app[i]->stop = true;
+    if (app[i]->handle) st20p_tx_wake_block(app[i]->handle);
+    pthread_join(app[i]->frame_thread, NULL);
+    info("%s(%d), sent frames %d(done %d)\n", __func__, i, app[i]->fb_send,
+         app[i]->fb_send_done);
+
+    tx_st20p_close_source(app[i]);
+  }
+
+  // check result
+  for (int i = 0; i < session_num; i++) {
+    if (app[i]->fb_send <= 0) {
+      err("%s(%d), error, no sent frames %d\n", __func__, i, app[i]->fb_send);
+      ret = -EIO;
+    }
+  }
+
+error:
+  for (int i = 0; i < session_num; i++) {
+    if (app[i]) {
+      if (app[i]->handle) st20p_tx_free(app[i]->handle);
+      free(app[i]);
+    }
+  }
+
+  /* release sample(st) dev */
+  if (ctx.st) {
+    mtl_uninit(ctx.st);
+    ctx.st = NULL;
+  }
+  return ret;
```

Comment:
Add an empty line before return
```suggestion

  return ret;
```